src/manager.rs

```diff
@@ -0,0 +1,212 @@
+    /// If the database file does not exist, it will be created.
+    ///
+    /// NOTE: You MUST consume the contents of `tx` to prevent memory leaks.
+    ///
+    /// The database stores the raw script source, but only after the source has been parsed
+    /// to ensure validity.
+    pub fn new(env: Env, path: &Path, tx: Box<T>) -> Result<Self, Error> {
+
+        let connection = try!(rusqlite::Connection::open(&path));
+        try!(connection.execute("CREATE TABLE IF NOT EXISTS scripts (
+            id          TEXT NOT NULL PRIMARY KEY,
+            source      TEXT NOT NULL,
+            is_enabled  BOOL NOT NULL DEFAULT 1
+        )", &[]));
+
+        Ok(ScriptManager {
+            path: path.to_owned(),
+            env: env,
+            runners: HashMap::new(),
+            tx: tx
+        })
+    }
+
+    /// Load and launch all existing scripts from the database.
+    pub fn load(&mut self) -> Result<HashMap<Id<ScriptId>, Result<(), Error>>, Error> {
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        let mut result_map = HashMap::new();
+        let mut stmt = try!(connection.prepare("SELECT id, source, is_enabled FROM scripts"));
+        let rows = try!(stmt.query(&[]));
+
+        for result_row in rows {
+            let row = try!(result_row);
+            let id_string: String = try!(row.get_checked(0));
+            let id: Id<ScriptId> = Id::new(&id_string);
+            let source: String = try!(row.get_checked(1));
+            let is_enabled: bool = try!(row.get_checked(2));
+
+            if is_enabled {
+                result_map.insert(
+                    id.clone(),
+                    self.start_script(&id, &source).map(|_| ()));
+            }
+        }
+        Ok(result_map)
+    }
+
+    /// Attempt to add a new script. The script will be executed and persisted to disk.
+    /// The ID is chosen by the consumer and must be unique.
+    pub fn put(&mut self, id: &Id<ScriptId>, source: &String) -> Result<(), Error> {
+        try!(self.start_script(&id, &source));
+
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        connection.execute("INSERT OR REPLACE INTO scripts (id, source, is_enabled)
+                VALUES ($1, $2, $3)", &[&id.to_string(), source, &1])
+            .map(|_| ()).map_err(From::from)
+    }
+
+    /// Enable or disable a script, starting or stopping the script if necessary.
+    pub fn set_enabled(&mut self, id: &Id<ScriptId>, enabled: bool) -> Result<(), Error> {
+        let source = try!(self.get_source(id));
+        let is_running = self.runners.contains_key(&id);
+        match (enabled, is_running) {
+            (false, true) => {
+                if let Some(mut runner) = self.runners.remove(&id) {
+                    let (tx, rx) = channel();
+                    runner.stop(move |result| {
+                        tx.send(result).unwrap();
```

Comment:
Please don't `unwrap()`. In that case, we just don't care if the channel has been closed, so `let _ = tx.send(result);`.